presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConnectorFactory.java

```diff
@@ -69,10 +66,10 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
                         binder.bind(NodeManager.class).toInstance(context.getNodeManager());
 
                         if (tableDescriptionSupplier.isPresent()) {
```

Comment:
Implementation of `ElasticsearchTableDescriptionProvider` is the same in both case of this `if`. Maybe instead of injecting the instance of `ElasticsearchTableDescriptionProvider` you could just inject `config`? Or better maybe you could provide proper `elasticsearch.properties`? Then this if would be not needed at all.

Reply:
yes, the implementation is the same.
I feels like this (if ... else) is introduced by:
requires a Plugin with no argument to load all plugins. So all production cases are running with tableDescrptionSupplier not present.
but testcases queryRunner needs customized plugin to run integration test. So has to provide 2 constructors, one with no argument, and one with a customized argument. My guess this is probably the reason why many connectors are doing similar things.
If exposing to elasticsearch.properties, it would be some configuration like: run-env, with value production or testing. I am not inclined to do it that way. 
What do you think?

Reply:
get it. thank you @kokosing @nezihyigitbasi 
I will have a test ElasticsearchConnectorFactory, and cleanup the code in Plugin and Factory